packages/tests.backcompat/src/client-backcompat.ts

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright 2025 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import chalk from "chalk";
+import consola from "consola";
+import { execa } from "execa";
+import { findUpSync } from "find-up";
+import { promises as fs } from "fs";
+import * as path from "node:path";
+import { fileURLToPath } from "node:url";
+import semver from "semver";
+
+const THIS_FILE_DIR = path.dirname(fileURLToPath(import.meta.url));
+const pnpmLockFile = findUpSync("pnpm-lock.yaml");
+
+if (!pnpmLockFile) {
+  throw new Error("pnpm-lock.yaml not found. :(");
+}
+
+const rootPath = path.dirname(pnpmLockFile);
+
+const monorepoTranspilePath = path.join(
+  rootPath,
+  "packages",
+  "monorepo.tool.transpile",
+  "bin",
+  "transpile2.mjs",
+);
+
+const builtSdksPath = path.join(
+  THIS_FILE_DIR,
+  "codegen",
+  "sdks",
+);
+
+async function codegenAtVersion(version: string, outPath: string) {
+  consola.info(
+    `Running codegen for version ${chalk.blue(version)}`,
+  );
+  const outSrcPath = path.join(outPath, "src");
+  await fs.mkdir(outSrcPath, { recursive: true });
+  const { failed } = await execa("pnpm", [
+    "client.test.generate",
+    "--o",
+    outSrcPath,
+    "--v",
+    version,
+  ]);
+
+  const testOntologyPackagePath = path.join(
+    rootPath,
+    "packages",
+    "client.test.ontology",
+  );
+
+  await fs.copyFile(
+    path.join(testOntologyPackagePath, "package.json"),
+    path.join(outPath, "package.json"),
+  );
+  await fs.copyFile(
+    path.join(testOntologyPackagePath, "tsconfig.json"),
+    path.join(outPath, "tsconfig.json"),
+  );
+  await fs.writeFile(
+    path.join(outPath, "pnpm-workspace.yaml"),
+    "",
+  );
+
+  await modifyPackageJsonToUseDirectPaths(path.join(outPath, "package.json"));
+
+  const pkgPath = path.join(outPath, "package.json");
+  const pkgData = JSON.parse(await fs.readFile(pkgPath, "utf8"));
+
+  /**
+   * Babel core is expecting these modules to be present in the workspace.
+   */
+  pkgData.dependencies = {
+    ...pkgData.dependencies,
+    "@babel/preset-typescript": "latest",
+    "@babel/preset-react": "latest",
+  };
+
+  await fs.writeFile(pkgPath, JSON.stringify(pkgData, null, 2));
+
+  await execa("pnpm", ["install"], { cwd: outPath });
+
+  /**
+   * monorepo.tool.transpile uses find-up on the pnpm-workspace.yaml file to find the root.
+   * Failing to remove it will cause the transpile command to fail.
+   */
```

Comment:
I think I'm confused by this too. What does the tool do when this file doesn't exist?